### ISSUE TITLE
Regression in output of sync_state_genSyncSpec

### ISSUE THREAD

**Issue by tomaka** [paritytech/substrate#11435]
### Is there an existing issue?

- [X] I have searched the existing issues

### Experiencing problems? Have you tried our Stack Exchange first?

- [X] This is not a support question.

### Description of bug

Before https://github.com/paritytech/substrate/pull/8783, calling the `sync_state_genSyncSpec` JSON-RPC function would return something like this:

```json
{"jsonrpc":"2.0","id":"...","result": {"name":"Westend", "chainType": "Live", ... } }
```

After the PR, the output looks like this:

```json
{"jsonrpc":"2.0","id":"...","result": "\{\"name\":\"Westend\", \"chainType\": \"Live\", ... \}" }
```

In other words, the result has been stringified.

**Comment by niklasad1** [comment]
oops, sorry let me have a look

**Comment by tomaka** [comment]
Note that I haven't actually checked what the function was returning before the PR (I have only checked the *after*). However we have a script that started failing and this script is directly expecting an object instead of a string, so what I wrote is the only possible explanation.

**Comment by niklasad1** [comment]
you are correct, it returned `serde_json::Value` before I think I was got annoyed that `as_json` already returned it as JSON but forgot that the Strings are serialized differently.

one sec, I think we need to block the current release for this right?
